### ISSUE TITLE
[no-duplicate-landmark-elements] False positive when using landmark elements within <dialog>

### ISSUE THREAD

**Issue by johanrd** [ember-template-lint/ember-template-lint#3152]
## Description

The `no-duplicate-landmark-elements` and `no-nested-landmark` rules in `ember-template-lint` currently flags valid uses of landmark elements (e.g., `<header>`, `<main>`, `<footer>`) when they are nested within a `<dialog>` element. This seems like a false positive because `<dialog>` creates a separate, modal context, and landmark elements within a dialog should not be considered duplicates of those outside the dialog.

## Steps to Reproduce

1.  Create a Glimmer component or template with a `<dialog>` element.
2.  Inside the `<dialog>`, include a landmark element like `<header>`.
3.  Also, include a `<header>` outside the dialog in the same template.
4.  Run `ember-template-lint`.

**Example Code:**

```html
<main>
  <header>
    <h1>Main Page Header</h1>
  </header>
  <button commandfor='my-dialog' command='show-dialog'>Open Dialog</button>
</main>

<dialog id='my-dialog'>
  <header>
    <h1>Dialog Header</h1>
  </header>
  <p>Some content in the dialog.</p>
</dialog>
```

## Expected Behavior:
The no-duplicate-landmark-elements rule should not flag the `<header>` inside the `<dialog>` as a duplicate. The same would perhaps also be relevant for html elements with the `popover` attribute

**Comment by lifeart** [comment]
Hi @johanrd! Thank you for detailed issue report!
Here is PR with a fix - https://github.com/ember-template-lint/ember-template-lint/issues/3152

**Comment by johanrd** [comment]
@lifeart wow, nice, thanks! tested locally, and can verify that your fix solves the problem💪
